packages/field-base/src/field-mixin.js

```diff
@@ -203,41 +243,26 @@ export const FieldMixin = (superclass) =>
       this._helperId = newId;
     }
 
-    /** @private */
-    __onHelperSlotChange() {
-      const current = this._currentHelper;
-
-      // Check fot slotted element node that is not the one created by this mixin.
-      const customHelper = this.__helperSlot.assignedElements({ flatten: true }).find((node) => node !== current);
-
-      if (customHelper) {
-        this.__updateHelperId(customHelper);
-
-        if (current && current.isConnected) {
-          current.remove();
-        }
-
-        this._applyCustomHelper();
-
-        this._currentHelper = customHelper;
-      }
-    }
-
     /** @protected */
-    _applyCustomHelper() {
-      const helper = this._helperNode.textContent;
-      if (helper !== this.helperText) {
-        this.helperText = helper;
+    _helperTextChanged(helperText) {
+      let helper = this._helperNode;
+
+      if (helperText) {
+        // Create helper lazily
+        if (!helper) {
+          helper = document.createElement('div');
+          helper.setAttribute('slot', 'helper');
+          this.__defaultHelper = helper;
+          this.__applyDefaultHelper(helper);
+        }
       }
-    }
 
-    /** @protected */
-    _helperTextChanged(helper) {
-      if (this._helperNode) {
-        this._helperNode.textContent = helper;
+      // Only set text content for default helper
+      if (helper && helper === this.__defaultHelper) {
+        helper.textContent = helperText;
       }
 
-      this.toggleAttribute('has-helper', Boolean(helper));
+      this.__toggleHasHelper(Boolean(helperText));
```

Comment:
For the custom helper, the `has-helper` attribute is added only when the **trimmed** helper text is not empty.

Should we do the same here?

Reply:
Good catch, updated.